src/main/java/com/visenze/productsearch/ProductSearchResponse.java

```diff
@@ -0,0 +1,295 @@
+package com.visenze.productsearch;
+
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.visenze.common.util.ViJsonMapper;
+import com.visenze.productsearch.response.ErrorMsg;
+import com.visenze.productsearch.response.GroupProductResult;
+import com.visenze.productsearch.response.Product;
+import com.visenze.visearch.ProductType;
+import com.visenze.visearch.Facet;
+import com.visenze.visearch.ResponseMessages;
+import com.visenze.visearch.internal.InternalViSearchException;
+import com.visenze.visearch.internal.http.ViSearchHttpResponse;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.visenze.visearch.internal.constant.ViSearchHttpConstants.BOX;
+import static com.visenze.visearch.internal.constant.ViSearchHttpConstants.SCORE;
+import static com.visenze.visearch.internal.constant.ViSearchHttpConstants.TYPE;
+
+/**
+ * <h1> ProductSearchResponse </h1>
+ * This class acts as a parser to help determine and break down the response
+ * generated by a HTTP request method stored inside ViHttpResponse.
+ *
+ * The @JsonIgnoreProperties(ignoreUnknown=true) annotation tells JSON that this
+ * class will ignore deserialization of fields found inside a json object that
+ * does not exists within our class (by default error is thrown as we are
+ * required to contain every field needed by the json object).
+ *
+ * <p>
+ * This class aims to be Json compatible by implementing Jackson annotation
+ *
+ * @author Shannon Tan
+ * @version 1.0
+ * @since 12 Jan 2021
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ProductSearchResponse extends ViJsonMapper {
+    /**
+     * Each request is associated with an id to help with tracking api calls.
+     *
+     * The actual field name is called 'reqid' but we want to store it as
+     * 'requestId' inside this class. The @JsonProperty("...") annotation tells
+     * JSON that this variable 'requestId' will be serialized/deserialized as
+     * 'reqid' from JSON's point of view.
+     *      i.e. some_json_map_thing["reqid"] = this_class.requestId;
+     */
+    @JsonProperty("reqid")
+    private String requestId;
+
+    /**
+     * The request status, either “OK”, “warning”, or “fail”.
```

Comment:
i believe the status is only either `OK` or `fail`
i dont rmb there is a `warning` status

Reply:
updated